### ISSUE TITLE
Using `sequential` underneath `parallel` causes tests to remain running in parallel

### ISSUE THREAD

**Issue by danbroooks** [hspec/hspec#438]
Looking at this issue here https://github.com/hspec/hspec/issues/311 the initial request describes wanting to have a top level `parallel` call for all tests to run in parallel by default, and then call `sequential` on some tests under this parallel call to mark some tests as unparallelisable. I note the implementation of `parallel` and `sequential` make use of `<|>` here:

https://github.com/hspec/hspec/blob/master/hspec-core/src/Test/Hspec/Core/Spec.hs#L140

So if we have already called `parallel` then `itemIsParallelizable item` is `Just True` and thus the result of `itemIsParallelizable item <|> Just value` when calling `sequential` inside a `parallel` block is:

```hs
Just True <|> Just False
> Just True
```

Was this intentional? Is it not possible for us to do:

```hs
setParallelizable :: Bool -> Item a -> Item a
setParallelizable value item = item {itemIsParallelizable = Just value}
```

Or will this not be as straightforward as that? Is the request in #311 possible? (Nested `parallel`/`sequential` calls)

**Comment by danbroooks** [comment]
> will this not be as straightforward as that? Is the request in #311 possible?

I just tried implementing my own `forceSequential` which does what i describe, and the answer is no. 

Is there no way then to achieve what #331 describes? Where specs are set to `parallel` by default, but some are marked as `sequential`, that are to be run sequentially, separate to specs marked as parallel?

**Comment by sol** [comment]
@danbroooks looking at https://github.com/hspec/hspec/blob/master/hspec-core/test/Test/Hspec/Core/SpecSpec.hs#L125, my initial assumption is that things work as expected.

Before we go into discussing any implementation details, please provide a minimal example + describe expected and actual behavior (as a new issue, I'm going to close this one as "invalid", not implying that there isn't a bug, just for the reason that I don't have a repro right now).
